### PR TITLE
Add PHP BCIs

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

File: src/bci_build/package.py [hunk 1]
```diff
@@ -1423,6 +1424,160 @@ def _get_openjdk_kwargs(
...
+            **extra_env,
+        },
+        extra_files={
+            "/usr/local/bin/docker-php-entrypoint": _php_entrypoint(php_variant),
+            "/usr/local/bin/docker-php-source": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-configure": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-enable": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-install": f"""#!/bin/sh
+set -eox pipefail
+zypper -n in --no-recommends php{php_version}-$1
```
Comment (dirkmueller) [inline src/bci_build/package.py:1565]: please install with recommends as that adds a couple of package-specified extra extensions

File: src/bci_build/package.py [hunk 2]
```diff
@@ -1423,6 +1424,160 @@ def _get_openjdk_kwargs(
...
+            "APACHE_CONFDIR": "/etc/apache2",
+            "APACHE_ENVVARS": "/usr/sbin/envvars",
+        }
+        cmd = ["apache2-foreground"]
+        custom_end = (
+            common_end
+            + """STOPSIGNAL SIGWINCH
+
+# create our own apache2-foreground from the systemd startup script
+RUN cat /usr/sbin/start_apache2 | sed 's|^exec $apache_bin|exec $apache_bin -DFOREGROUND|' > /usr/local/bin/apache2-foreground
```
Comment (dirkmueller) [inline src/bci_build/package.py:1477]: I wonder if we can do that by setting /etc/sysconfig/apache2 define. would have a slightly better optics

File: src/bci_build/package.py [hunk 3]
```diff
@@ -1423,6 +1424,160 @@ def _get_openjdk_kwargs(
...
+        + extra_pkgs,
+        replacements_via_service=[
+            Replacement("%%composer_version%%", package_name="php-composer2")
+        ],
+        cmd=cmd,
+        entrypoint=["docker-php-entrypoint"],
+        env={
+            "PHP_VERSION": php_version,
+            "PHP_INI_DIR": f"/etc/php{php_version}/",
+            "PHPIZE_DEPS": f"php{php_version}-devel autoconf gawk gcc make file",
```
Comment (dirkmueller) [inline src/bci_build/package.py:1554]: do we still need this? do we need to install those packages?

File: src/bci_build/package.py [hunk 4]
```diff
@@ -1423,6 +1424,160 @@ def _get_openjdk_kwargs(
...
+        cmd = ["php-fpm"]
+        custom_end = r"""WORKDIR /srv/www/htdocs
+
+RUN set -eux; \
+	cd /etc/php8/fpm/; \
+        cp php-fpm.d/www.conf.default php-fpm.d/www.conf; \
+        cp php-fpm.conf.default php-fpm.conf; \
+	{ \
+		echo '[global]'; \
+		echo 'error_log = /proc/self/fd/2'; \
```
Comment (dirkmueller) [inline src/bci_build/package.py:1500]: so this logs to stderr. does that work with apache2 doing that as well?

File: src/bci_build/package.py [hunk 5]
```diff
@@ -1423,6 +1428,163 @@ def _get_openjdk_kwargs(
...
+            "COMPOSER_VERSION": "%%composer_version%%",
+            **extra_env,
+        },
+        extra_files={
+            "/usr/local/bin/docker-php-entrypoint": _php_entrypoint(php_variant),
+            "/usr/local/bin/docker-php-source": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-configure": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-enable": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-install": f"""#!/bin/sh
+set -eox pipefail
```
Comment (dirkmueller) [inline src/bci_build/package.py:1571]: `set -euo pipefail`

File: src/bci_build/package.py (hunk 3 above)
Comment (dcermak) [reply src/bci_build/package.py:1566]: Not really I think

File: src/bci_build/package.py [hunk 6]
```diff
@@ -982,7 +986,7 @@ async def write_file_to_dest(fname: str, contents: Union[str, bytes]) -> None:
 
         for fname, contents in self.extra_files.items():
             files.append(fname)
-            tasks.append(asyncio.ensure_future(write_file_to_dest(fname, contents)))
+            tasks.append(write_file_to_dest(fname, contents))
```
Comment (dirkmueller) [inline src/bci_build/package.py:989]: whats this change about? is this intentional?

**Review by dirkmueller — DISMISSED** [review]
LGTM, question inline (on the asyncio.ensure* removal)

**Review by dirkmueller — DISMISSED** [review]

**Review by dirkmueller — APPROVED** [review]

**Pull request merged**